### DISCUSSION TITLE
support multiple root directories for file picker

### DISCUSSION THREAD

**Discussion by jyn514** [helix-editor/helix#11589] [Ideas]
often when i open a file outside the current workspace root, i'm opening multiple other files in the same directory as well. it would be nice to have a fuzzy-picker for them, the same way that ` f` opens a fuzzy picker for the original directory. otherwise i have to find the absolute path of the file to open and manually type that into `:open`.

**Comment by pascalkuthe** [comment]
I don't really understand what you are asking for.

You can use `:o <firectory>` to open a file picker in analog rbitrary directory and you could even create keybindings for that.

**Comment by jyn514** [comment]
i don't want to open a file picker for *only* a specific arbitrary directory though, i want a single unified file picker for every directory i have open. basically i want to merge the picker i get from `:o` and the picker i get from ` f` into a single picker that includes all the files from both.

**Comment by archseer** [comment]
You could change the current directory then use `space+F` which is based on cwd?

**Comment by jyn514** [comment]
that has the same downside though, now there are two distinct pickers i have to keep track of. i want helix to do the work for me, since it already know the paths of all the files i've opened.

**Comment by sdoerner** [comment]
This would be pretty useful for big monorepos where you only care about a subset of paths within that repo but they don't have a common root. Perhaps ":open" could be changed to accept multiple paths?

**Comment by sdoerner** [comment]
Currently, if you use `:open` with multiple directories, it will spawn multiple file pickers on top of each other. I have been testing a modified version where instead it opens a single picker with multiple roots. This has been working pretty well to address this monorepo use case.

One caveat is that the least common root path may be pretty far up in the file tree, leading to long file names in the picker. This might be mitigated by supporting some kind of configurable "alias" system for common prefixes (e.g. `java/com/foo/bar/bla` -> `j/../bla`) if that is your root package) in `editor.filepicker` which could shorten the displayed names, but this is out of scope of MVP.

Prior discussion of the current behavior was at https://github.com/helix-editor/helix/pull/2707 and the choice of "multiple pickers" seems deliberate though.

Possible ways to support this different workflow:
 1. change the default behavior of :open
 2. add a config option in `editor.filepicker`
 3. add a new command (maybe `:picker`?)

I think the original use case discussed at the top here seems a bit more specific, and perhaps not generic enough for mainline?
However, it could perhaps be supported with a plugin (once the plugin system exists) more easily once `:open` (or `picker`) is able to support multiple roots in the first place.

Would you be open to having any of the 3 options above (or any other options to support this) merged?
